### PR TITLE
Edit API doc

### PR DESCRIPTION
**Pull request by rkuchan**
This takes a stab at an API reference doc. It adds detail to the API doc in the Concepts section. It covers each endpoint (`/ping`,`query`, and `/write`) and, where applicable, goes into detail about the required and optional query string parameters, status codes and responses, and example requests. 

The sections on status codes and query string parameters need special attention - I included things I was able to test but I have a feeling I am missing some.

I would like this to live somewhere else on the docs but that discussion can occur offline as it involves a significant reorganization of the sidebar.

This fixes:
* https://github.com/influxdata/docs.influxdata.com/issues/454 (Clarify batch size recommendations)
* https://github.com/influxdata/docs.influxdata.com/issues/441 (Clarify `GET`/`POST` for the `/query` endpoint)
* https://github.com/influxdata/docs.influxdata.com/issues/442 (Clean up write syntax) - I don't actually touch the write syntax page in this PR but a sidebar reorganization would make this the main API doc.

### CODE REVIEW

File: content/influxdb/v0.13/concepts/api.md [hunk 1]
```diff
@@ -26,32 +51,341 @@ X-Influxdb-Version: 0.13.x
...
+| rp=\<retention_policy_name> | Optional | Sets the target [retention policy](/influxdb/v0.13/concepts/glossary/#retention-policy-rp) for the query. InfluxDB queries the `DEFAULT` retention policy if you do not specify a retention policy.  |
+| u=\<username> | Optional if you haven't [enabled authentication](/influxdb/v0.13/administration/authentication_and_authorization/#set-up-authentication). Required if you've enabled authentication. | Sets the username for authentication if you've enabled authentication. The user must have read access to the database. Use with the query string parameter `p`. |
+
+#### Request Body
+
+```
+--data-urlencode "q=<InfluxQL query>"
+```
+
+Use curl's `--data-urlencode` encoding method for all queries.
```
Comment (beckettsean) [inline content/influxdb/v0.13/concepts/api.md]: URL encoding isn't something specific to `curl`. Let's reword to something like:

All queries myst be URL encoded and in [InfluxQL](influxdb/v0.13/query_language/), InfluxDB's query language. 
Our example show's the `--data-urlencode` parameter from `curl`, which we will use in all examples on this page.

File: content/influxdb/v0.13/concepts/api.md [hunk 2]
```diff
@@ -26,32 +51,341 @@ X-Influxdb-Version: 0.13.x
...
+| u=\<username> | Optional if you haven't [enabled authentication](/influxdb/v0.13/administration/authentication_and_authorization/#set-up-authentication). Required if you've enabled authentication. | Sets the username for authentication if you've enabled authentication. The user must have write access to the database. Use with the query string parameter `p`. |
+
+
+#### Request Body
+
+```
+--data-binary '<Data in Line Protocol format>'
+```
+
+Use curl's `--data-binary` encoding method for all writes.
```
Comment (beckettsean) [inline content/influxdb/v0.13/concepts/api.md]: Similar to above, binary encoding isn't specific to `curl`. 

All line protocol writes must be binary encoded and in the [Line Protocol format](https://docs.influxdata.com/influxdb/v0.13/write_protocols/line/).
Our example shows the `--data-binary` parameter from curl, which we will use in all examples on this page.

File: content/influxdb/v0.13/concepts/api.md [hunk 3]
```diff
@@ -26,32 +51,341 @@ X-Influxdb-Version: 0.13.x
...
-### Admin interface port
+##### Summary Table
+<br>
 
-The admin interface for InfluxDB runs on port `8083` and exposes web UI for the server.
+| HTTP status code | Description    |
+| :--------------- | :------------- |
+| 204 No Content   | Success!      |
+| 400 Bad Request  | Unacceptable request. Can occur with a Line Protocol syntax error or if a user attempts to write values to a field that previously accepted a different value type. The returned JSON offers further information. |
+| 404 Not Found    | Unacceptable request. Can occur a user attempts to write to a database that does not exist. The returned JSON offers further information. |
```
Comment (beckettsean) [inline content/influxdb/v0.13/concepts/api.md]: Can occur **if** a user attempts...

**Comment by beckettsean** [comment]
great doc!

**Comment by rkuchan** [comment]
Thanks, @beckettsean!

**Pull request merged**